ci_example_cpp/doc/cpp_coding_guidelines.md

```diff
@@ -0,0 +1,270 @@
+    g_, i.e. *g_variable_name*.
+
+## IV. Add Units to Variable Names
+
+Variables that hold values of a specific unit should have that unit appended to
+the name.  For example if a variable holds the velocity of a motor in *krpm* it
+should be called `velocity_krpm` instead of `just velocity`. Some more examples:
+
+- duration_us (use "u" instead of "µ")
+- voltage_mV
+- acceleration_mps2 (\f$ \frac{m}{s^2} \f$)
+
+## V. C/C++ Formatting
+
+### V.1. Line Length
+
+Limit the length of lines to 80 characters.
+
+This may seem hard to follow sometimes but makes it much easier to view two or even three files next to each other (important during code review or when resolving merge conflicts).
+
+### V.2. Indentation
+
+- Use spaces instead of tabs.
+- 4 spaces per "tab".
+
+### V.3. Position of braces
+
+- Opening brace always goes to the next line.
+- **Always** add braces for single-line if/loop/etc.
+
+Example:
+
+~~~{.c}
+namespace bar
+{
+
+class Foo {
+    void my_function(const Foo &foo, int *output_arg)
+    {
+        if (condition)
+        {
+            ...
+        }
+        else if (other_condition)
+        {
+            ...
+        }
+        else
+        {
+            ...
+        }
+
+        while (condition)
+        {
+            ...
+        }
+    }
+}
+
+} // namespace
+~~~
+
+### V.3. Spaces
+
+Add single spaces between if/for/etc., the condition and the brace. add spaces
+around most binary operators. Exception: No spaces around *::*, *.* and *->*.
+Also no spaces for unary operators (*i++*, *&x*, **x*, ...).
+
+Example:
+~~~{.c}
+int x = 42;
+for (int i = 0; i < x; i++)
+{
+    int y = i * x;
+    if (y == foo.bar->baz)
+    {
+        ...
+    }
+}
+~~~
+
+### V.4. Formatting of switch blocks
+
+- See the following example for proper indentation of switch blocks.
+- Always add a default case, even when it is technically not needed.
+    - If the default block is empty, it shall contain a comment to indicate that
+    this is intentional.
+    - The default case shall either be the first or the last case, preferably the
+    last.
+- Non-empty cases shall be terminated by an unconditional break.
+
+~~~{.c}
+switch (x)
+{
+    case 1:
+        // ...
+        break;
+ 
+    case 2:
+        // ...
+        break;
+ 
+    case 3: // multiple cases for one block are okay
+    case 4:
+        // ...
+        break;
+ 
+    case 5:           // BAD. A non-empty case without break should not be
+        something();  // used
+    case 6:
+        more();
+        break;
+ 
+    default:
+        // no action needed
+        break;
+}
+~~~
+
+### V.5. Clang-Format Configuration
+
+To automatically format your code according to this guidelines, you can use
+clang-format with the following configuration:
+
+~~~
+BasedOnStyle: Google
+AccessModifierOffset: '-4'
+AllowShortFunctionsOnASingleLine: None
+BinPackArguments: 'false'
+BinPackParameters: 'false'
+BreakBeforeBraces: Allman
+IndentWidth: '4'
+~~~
```

Comment:
Now that we have the config file in mpi_cmake_modules, we should link to the file there instead of pasting it here.